lib/rex/proto/dns/server.rb

```diff
@@ -222,7 +223,7 @@ def running?
   #
   # Start the DNS server and cache
   # @param start_cache [TrueClass, FalseClass] stop the cache
-  def start(start_cache = true)
+  def start
```

Comment:
Should this be considered a breaking change since custom modules may rely on this signature.

Reply:
Fair question. The issue I have with this line of code is that it is different to the other services started by the service manager, so consistent handling of services is made more difficult. It puts a user of this code in a position where they can choose to either use that parameter, or start it through the service manager; I'm not sure how you do both cleanly. Actually, the SSH server also has an optional parameter in `start`, which far as I can tell isn't used anywhere; but it would have the same problem if anyone did want to use it.

We could keep the interface effectively the same by leaving the optional parameter in the `start` method, and similarly, making the new value in the constructor optional as well, and then emulating the old behaviour in the case where neither of these parameters are set... I don't love how messy that starts to get, but if breaking changes for code outside this repository are important, it's a way of solving it. Happy to take your guidance as to the tradeoff.

Reply:
I think that this should be fine - i clearly messed up in the calling convention translation up into Msf:: since the intent _was_ to give users access to cache semantics since that's how spoofer and such work.